sources/packages/api/src/services/sfas/sfas-part-time-application.service.ts

```diff
@@ -46,6 +46,51 @@ export class SFASPartTimeApplicationsService
     await this.repo.save(application, { reload: false, transaction: false });
   }
 
+  /**
+   * Validates before an application submission to see if there is an overlapping SFAS part-time application existing.
+   * @param sin
+   * @param birthDate
+   * @param lastName
+   * @param studyStartDate
+   * @param studyEndDate
+   * @returns SFAS part-time application.
+   */
+  async validateDateOverlap(
+    sin: string,
+    birthDate: Date,
+    lastName: string,
+    studyStartDate: Date,
+    studyEndDate: Date,
+  ): Promise<SFASPartTimeApplications> {
+    return this.repo
+      .createQueryBuilder("sfasPTApplication")
+      .select(["sfasPTApplication.id"])
+      .innerJoin("sfasPTApplication.individual", "sfasPTstudent")
+      .where("sfasPTstudent.lastName = :lastName", { lastName })
+      .andWhere("sfasPTstudent.sin = :sin", { sin })
+      .andWhere("sfasPTstudent.birthDate = :birthDate", { birthDate })
+      .andWhere(
+        new Brackets((qb) => {
+          qb.where(
+            "sfasPTApplication.startDate BETWEEN :startDatePT AND :endDatePT",
+            { startDatePT: studyStartDate, endDatePT: studyEndDate },
+          )
+            .orWhere(
+              "sfasPTApplication.endDate BETWEEN :startDatePT AND :endDatePT",
+              { startDatePT: studyStartDate, endDatePT: studyEndDate },
+            )
+            .orWhere(
+              "startDatePT: BETWEEN sfasPTApplication.startDate AND sfasPTApplication.endDate",
```

Comment:
Please check the parameter syntax for "startDatePT:" on line 83. The ":" seems to be in the wrong place, same for line 87.

Reply:
fixed.